### PR TITLE
Start refactor

### PR DESCRIPTION
**Pull request by awsmsrc**
## Release Playbook
- [ ] Rebase against master
- [ ] Release branch ()
- [ ] Pass CI ()
- [ ] Code review
- [ ] Merge into master
- [ ] Release master ()
- [ ] Pass CI ()
- [ ] Update staging rack
- [ ] Edit [Rack release record](https://github.com/convox/rack/releases) and/or update [docs](https://github.com/convox/convox.github.io)
- [ ] Publish release
- [ ] Release CLI

### CODE REVIEW

**Comment by awsmsrc** [comment]
- [ ] Check min docker version
- [ ] #763 validate cron labels
- [ ] #762 take .dockerignore into account when using sync
- [ ] #757 use docker archiving tool
- [ ] #742 dont report bad yaml errors on start

**Comment by ddollar** [comment]
- [ ] #813 Handle relative destination directories for code sync
- [ ] #812 Handle builds as map as well as string

**Comment by ddollar** [comment]
- [ ] #821 long output lines break convox start

File: cmd/convox/start_test.go [hunk 1]
```diff
@@ -41,82 +32,82 @@ redis:
...
-		test.ExecRun{
-			Command:  fmt.Sprintf("convox start"),
-			Dir:      appDir,
-			Env:      map[string]string{"CONVOX_CONFIG": temp},
-			OutMatch: "REDIS_URL",
-			Exit:     0,
-		},
-	)
-}
+// func TestStartWithMissingEnv(t *testing.T) {
```
Comment (nzoschke) [inline cmd/convox/start_test.go]: These can be deleted if they aren't meaningful moving forward.

File: manifest/manifest_test.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+	"testing"
+	"time"
+
+	"github.com/convox/rack/manifest"
+	"github.com/stretchr/testify/assert"
+)
+
+// WARNING: make sure to use spaces for the yaml indentations
+
+func TestLoadVersion1(t *testing.T) {
```
Comment (nzoschke) [inline manifest/manifest_test.go]: These are great!

File: manifest/proxy.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	Balancer  int
+	Container int
+
+	Protocol string
+	Host     string
+	Proxy    bool
+	Secure   bool
+}
+
+func (p *Proxy) Start() error {
```
Comment (nzoschke) [inline manifest/proxy.go]: I've observed the proxy container hanging around on `convox start` crashes. Can we address this with a defer or something?

**Comment by nzoschke** [comment]
I did a code review and LGTM except for the commented out tests.

It's working just fine here in testing too. Ship it! 🚀

**Comment by ddollar** [comment]
Merged in #882

**Pull request closed without merging**